compiler/modules/nback/equal.bal

```diff
@@ -190,3 +206,17 @@ function buildEqualStringString(llvm:Builder builder, Scaffold scaffold, CmpEqOp
     }
     buildStoreBoolean(builder, scaffold, b, result);
 }
+
+function buildEqualDecimalDecimal(llvm:Builder builder, Scaffold scaffold, boolean exact, CmpEqOp op, llvm:PointerValue tagged1, llvm:PointerValue tagged2, bir:Register result) {
+    llvm:Value b;
+    if exact {
+        b = <llvm:Value>builder.call(scaffold.getRuntimeFunctionDecl(decimalExactEqFunction), [tagged1, tagged2]);
+    }
+    else {
+        b = builder.iCmp("eq", <llvm:Value>builder.call(scaffold.getRuntimeFunctionDecl(decimalCmpFunction), [tagged1, tagged2]), llvm:constInt(LLVM_INT, 0));
+    }
+    if op == "ne" {
```

Comment:
Use this only in the exact case. For the other case, pass the right operator to iCmp.

Reply:
Changed